documentation/decisions/0007-model-resolution.md

```diff
@@ -0,0 +1,102 @@
+  are all defined in the same namespace.
+* Definition 10: A _models root_ is an identifier for a place where aspect model files are stored.
+  It could be a file system folder, a URL of a GitHub repository, a SPARQL endpoint URL or similar.
+* Definition 11: A _resolution strategy_ is a function that takes a models root and an Aspect Model
+  Element Identifier as input and returns the model element definition for the model element
+  identified by the Aspect Model Element Identifier, including information about the aspect model
+  file that contains the model element definition; or indicates a resolution failure (i.e., no model
+  element definition exists in this models root). It is allowed that the resolution strategy can be
+  configured with additional configuration (such as credentials), but this configuration should not
+  be required to be given in each function call.
+* Definition 12: An _aspect model resolver_ is a function that takes at least one models root, at
+  least one resolution strategy and at least one Aspect Model Element Identifier as input and
+  returns model element definitions for all transitively referenced model elements. Note that model
+  elements referred to by samm:see are not considered referenced in an aspect model resolver.
+
+### Restrictions - Rules for files, namespaces and model elements
+
+* An aspect model file may only contain model elements of exactly one namespace.
+* Model element definitions of a namespace may be spread across any number of aspect model files
+  (i.e., the namespace's namespace files).
+* A model element definition may only appear once per namespace and aspect model file.
+* A namespace may contain any number of model elements, including `samm:Aspect`.
+* An aspect model file may contain at most one model element definition for samm:Aspect.
+* A namespace's set of namespace files may contain at most one namespace element definition, which must match the namespace.
+* In filesystem-based model storage, aspect model files must end with ".ttl".
+* In filesystem-based model storage, models should be stored in the directory structure
+  _models_root_/_package_part_/_version_part_/AspectModelFile.ttl.
+* In filesystem-based model storage, the namespace file containing an Aspect model element
+  definition should have the same name as the Aspect's local name suffixed with ".ttl".
+  
+### Resolution of files
+
+![Model structure for loaded models](0007-model-resolution.svg)
+
+* While the aspect model resolver is responsible for resolving single model element definitions and
+  therefore is considered an internal API, the public API to "load" an Aspect Model should return an
+  instance of the "AspectModel" interface as shown above.
+* The `AspectModel` has information about its contained namespaces, and the files it consists of
+  (the `ModelFile` abstraction corresponds to the aspect model file concept described in the above
+  section).
+
```

Comment:
- Very precision definitions. Are more examples useful? Probably not required given the audience.
- Should ModelElementGroup (see graphic) also become a definition?

Reply:
Added _model element group_ definition.